### PR TITLE
VPN-5145: Implement Mobile onboarding flow

### PR DESCRIPTION
**Pull request by MattLichtenstein**
## Description

- Implements the onboarding flow for iOS and Android
  - Different "Start" slide than desktop which prompts the user to allow/disallow adding the VPN configuration to the system
  - Various UI layout changes to support the other cross-platform slides
- Moves onboarding after device limit screen ([see comment](https://github.com/mozilla-mobile/mozilla-vpn-client/pull/8141/files#diff-a131ac478dadc38997760e67ddb60a6657adc7fb3790e77542c3bce63b659268R405))
- Replace old onboarding QR codes with styled QR codes

https://github.com/mozilla-mobile/mozilla-vpn-client/assets/15353801/0a572fea-2889-4a50-ac7d-ffe15782ed16

## Reference

[VPN-5145: Implement Mobile onboarding flow](https://mozilla-hub.atlassian.net/browse/VPN-5145)
[VPN-5631: [a11y] Users are able to move forward through the new onboarding flow while using the progress bar when the Screen reader is enabled ](https://mozilla-hub.atlassian.net/browse/VPN-5631)

### CODE REVIEW

File: src/translations/strings.yaml
Comment (flodolo) [inline src/translations/strings.yaml:559]: The new label feels quite confusing: any explanation from content about the reason for this change?

File: src/translations/strings.yaml
Comment (flodolo) [inline src/translations/strings.yaml:579]: > tablet or mobile device’s camera 

Non-native speaker: shouldn't both have the genitive?

File: src/translations/strings.yaml
Comment (traceyvantyghem) [reply src/translations/strings.yaml:559]: @flodolo Agreed. I think this was copied over from the desktop onboarding, which has as the last step "Startup" and prompts users to turn on "Connect VPN on startup". 

This is a totally different step, though. It seems like a better label here might be "Connect". Thoughts? (If @flodolo is ok with this change, @MattLichtenstein is there an issue with making this update?)

File: src/translations/strings.yaml
Comment (traceyvantyghem) [reply src/translations/strings.yaml:579]: @flodolo Well, the genitives look ok to me, but I think this sentence could be a bit more direct.

What about "To install the VPN on another tablet or mobile device, scan the QR code."?

File: src/translations/strings.yaml
Comment (flodolo) [reply src/translations/strings.yaml:559]: `Connect` works for me, there's no space for interpretation

File: src/translations/strings.yaml
Comment (flodolo) [reply src/translations/strings.yaml:579]: Works fine for me 👍🏼

File: src/translations/strings.yaml
Comment (MattLichtenstein) [reply src/translations/strings.yaml:559]: No strong opinions here, but for a bit of background: 

The step was originally labelled "Get started" to denote it was the last step. This eventually got shortened to "Start" and it seems like that meaning may have gotten lost in translation when we switched the final slide on desktops content to being about connecting on startup (instead of quick access), leading to the final step being called "Startup." This now has a different meaning than the intended "Get started" and I agree it doesn't make much sense on mobile. I prefer "Connect" but also don't mind going back to "Start" if we wanted to be consistent across platforms (which I don't think is absolutely necessary, since the content IS different, just another option)

File: src/ui/screens/onboarding/OnboardingPrivacySlide.qml
Comment (mcleinman) [inline src/ui/screens/onboarding/OnboardingPrivacySlide.qml:44]: Is there a utility function available for `Qt.platform.isMobile`? If not, is it worth making one?

Comment (mcleinman) [inline src/ui/screens/onboarding/OnboardingPrivacySlide.qml:44]: Is there a utility function available for `Qt.platform.isMobile`? If not, is it worth making one?

File: src/mozillavpn.cpp
Comment (mcleinman) [inline src/mozillavpn.cpp:416]: Is this bug being resolved in another ticket?

File: src/mozillavpn.cpp
Comment (mcleinman) [inline src/mozillavpn.cpp:405]: Thanks for this comment, super helpful for future maintenance.

File: src/mozillavpn.cpp
Comment (mcleinman) [inline src/mozillavpn.cpp:136]: Why this update? Because we need to be able to start the connection during onboarding now? Why didn't we need this for desktop, out of curiousity?

File: src/platforms/ios/ioscontroller.swift
Comment (mcleinman) [inline src/platforms/ios/ioscontroller.swift:192]:
```suggestion
                        IOSControllerImpl.logger.error(message: "Finishing onboarding, but saving tunnel resulted in error...")
```

File: src/platforms/ios/ioscontroller.swift
Comment (mcleinman) [inline src/platforms/ios/ioscontroller.swift:202]: Should this (and the above one) be logged as an error or another level?

File: android/vpnClient/src/main/java/org/mozilla/firefox/vpn/qt/VPNActivity.java
Comment (mcleinman) [inline android/vpnClient/src/main/java/org/mozilla/firefox/vpn/qt/VPNActivity.java:165]: `cancelled` for this (in several files) almost makes it sound like the user cancelled something. Did you consider something like `EVENT_ONBOARDING_COMPLETE` (that's a little long, but hopefully you get my gist)?

**Review by mcleinman — COMMENTED** [review]
Just a few questions, mostly so I better understand this. Thanks! It's looking great!

File: src/mozillavpn.cpp
Comment (MattLichtenstein) [reply src/mozillavpn.cpp:416]: [VPN-5520: All pre-onboarding glean data being deleted on first sign in](https://mozilla-hub.atlassian.net/browse/VPN-5520)

File: src/mozillavpn.cpp
Comment (MattLichtenstein) [reply src/mozillavpn.cpp:136]: With the current implementation, we need the connection manager initialized in onboarding as a means to present the "Add VPN configuration" system modal (starting with `ConnectionManager::activate()`). We do not have this requirement on desktop.

File: src/platforms/ios/ioscontroller.swift
Comment (MattLichtenstein) [reply src/platforms/ios/ioscontroller.swift:192]: Do you think this clarification is still needed after the error log 3 lines above that says "Connect Tunnel Save Error"?

File: android/vpnClient/src/main/java/org/mozilla/firefox/vpn/qt/VPNActivity.java
Comment (MattLichtenstein) [reply android/vpnClient/src/main/java/org/mozilla/firefox/vpn/qt/VPNActivity.java:165]: my original reasoning was to make the naming for all of this generic enough so that it made sense if the reasoning was something other than onboarding in the future. as you can see, I eventually went against that on the iOS side, so I will make it more consistent on Android as well

File: src/translations/strings.yaml
Comment (traceyvantyghem) [reply src/translations/strings.yaml:559]: I think, since this really is a different step from the desktop onboarding, it's better to be more specific here than consistent across platforms. (My understanding is that these labels are acting as navigation labels for screen readers, and "Connect" is definitely a more clear and accurate label for this step than "Start").

File: src/translations/strings.yaml
Comment (traceyvantyghem) [reply src/translations/strings.yaml:579]: @MattLichtenstein sorry for the delay - yes, let's update on desktop too. Thank you!

File: src/mozillavpn.cpp
Comment (mcleinman) [reply src/mozillavpn.cpp:422]: We decided not to block on this, right?

File: src/ui/screens/onboarding/OnboardingPrivacySlide.qml
Comment (mcleinman) [reply src/ui/screens/onboarding/OnboardingPrivacySlide.qml]: Thanks!

File: src/platforms/ios/ioscontroller.swift
Comment (mcleinman) [reply src/platforms/ios/ioscontroller.swift]: Personally, I think it makes it even clearer when a future person is skimming logs, but I don't feel strongly about that at all - I'm okay if you want to keep it as is.

File: src/mozillavpn.cpp
Comment (mcleinman) [reply src/mozillavpn.cpp]: Makes sense, thanks.

**Review by mcleinman — CHANGES_REQUESTED** [review]
After logging in on iOS, this is the first thing I see:
IMG_D30F33317E6B-1

**Comment by MattLichtenstein** [comment]
> After logging in on iOS, this is the first thing I see:
> 
> IMG_D30F33317E6B-1
> 
> 

This onboarding experience is still behind the `newOnboarding` feature flag

File: src/mozillavpn.cpp
Comment (MattLichtenstein) [reply src/mozillavpn.cpp:422]: We are blocking turning the feature flag on, on that ticket

**Comment by mcleinman** [comment]
> > After logging in on iOS, this is the first thing I see:
> > [removed big image]
> 
> This onboarding experience is behind the `newOnboarding` feature flag, which is currently off by default

But with or without the feature flag, this isn't something that should be seen on mobile, right?

**Comment by MattLichtenstein** [comment]
> > > After logging in on iOS, this is the first thing I see:
> > > [removed big image]
> > 
> > 
> > This onboarding experience is behind the `newOnboarding` feature flag, which is currently off by default
> 
> But with or without the feature flag, this isn't something that should be seen on mobile, right?

🤦‍♂️ yup, totally misinterpreted the comment. now fixed

**Review by mcleinman — APPROVED** [review]
👍🏻 Looks great, tests great on iOS. (I had to turn on the feature manually; I believe that is expected.) Thanks!

File: src/translations/strings.yaml
Comment (flodolo) [reply src/translations/strings.yaml:559]: I think this is unclear also on Desktop. Does it mean "Start the VPN" or something else? I would have no clue how to translate this without additional context; without it, we're going to get very confusing translations.

File: src/translations/strings.yaml
Comment (flodolo) [inline src/translations/strings.yaml:582]: Nitpicking, but there is no Mozilla VPN website, there's a mozilla.org website with VPN pages.

File: src/translations/strings.yaml
Comment (flodolo) [inline src/translations/strings.yaml:592]: Why the switch to approve, if the terminology has been "allow" so far (and probably is in the phone)?

> Your mobile device will now ask you to allow us to create a VPN connection. Please allow this so that Mozilla VPN works correctly on your device.

I find it very confusing when software uses "we" to refer to itself. We're already repeating "allow" a lot, it doesn't seem so bad to repeat "Mozilla VPN".

> Your mobile device will now ask you to allow Mozilla VPN to create a VPN connection. Please allow this so that Mozilla VPN works correctly on your device.

**Review by flodolo — CHANGES_REQUESTED** [review]
Mostly nitpicking on the language, but the "Startup" string needs a clarification also for desktop.

File: src/translations/strings.yaml
Comment (MattLichtenstein) [reply src/translations/strings.yaml:592]: agree with all of this! @traceyvantyghem thoughts?

File: src/translations/strings.yaml
Comment (traceyvantyghem) [reply src/translations/strings.yaml:582]: Ok, I would leave this one, because you can go to Google and search "Mozilla VPN", and you do get what looks like a dedicated site as the first hit. I know it's technically pages within the Mozilla site, but being more accurate here doesn't really seem like it'll add any value, imo.

**Comment by flodolo** [comment]
Could you provide a screenshot? I feel like I'm getting more confused at this point :-(

**Comment by traceyvantyghem** [comment]
> Could you provide a screenshot? I feel like I'm getting more confused at this point :-(

Sorry @flodolo - I totally misinterpreted your comment. My mistake. Matt reached out in Slack to let me know about the confusion and said we should be good on the Startup/Connect question.

**Comment by flodolo** [comment]
> Sorry @flodolo - I totally misinterpreted your comment. My mistake. Matt reached out in Slack to let me know about the confusion and said we should be good on the Startup/Connect question.

No worry, the confusion is actually unrelated to your last comments. I feel like I'm not fully understanding how this label is used, and that's necessary to evaluate the comment.

**Comment by MattLichtenstein** [comment]
> No worry, the confusion is actually unrelated to your last comments. I feel like I'm not fully understanding how this label is used, and that's necessary to evaluate the comment.

@flodolo I will be adding a better comments to these labels in an upcoming commit

File: src/translations/strings.yaml
Comment (traceyvantyghem) [reply src/translations/strings.yaml:592]: Agree about "Allow".

I think we could get away from using "us" here, but maybe we could avoid a little of the repetition. What about this:

**Allow Mozilla VPN to create a connection**

Your mobile device will now ask you to allow Mozilla VPN to create a connection. Please allow this so that the VPN works correctly on your device.

[Connect Mozilla VPN]

**Review by flodolo — APPROVED** [review]

**Pull request merged**
